packages/client/src/components/SafeTokens/index.js

```diff
@@ -0,0 +1,54 @@
+import { useContext } from "react";
+import { useLocation, useHistory } from "react-router-dom";
+import { Web3Context } from "contexts/Web3";
+import { useModalContext } from "contexts";
+import { SendTokens } from "components";
+import VaultTable from "./VaultTable";
+import EmptyVault from "./EmptyVault";
+import { ASSET_TYPES } from "constants/enums";
+const SafeTokens = () => {
+  const location = useLocation();
+  const history = useHistory();
+  const { openModal } = useModalContext();
+  const treasuryAddress = location.pathname.split("/")[2];
+  const web3 = useContext(Web3Context);
+  const balances = web3?.balances?.[treasuryAddress] ?? {};
+  const vaults = Object.entries(balances).map(([coinType, balance]) => ({
+    coinType,
+    balance,
+  }));
+  const handleSendToken = (coinType) => {
+    openModal(
+      <SendTokens
+        address={treasuryAddress}
+        initialState={{
+          assetType: ASSET_TYPES.TOKEN,
+          coinType,
+        }}
+      />
+    );
+  };
+  const handleManageTokenVaults = () => {
+    history.push(`/safe/${treasuryAddress}/settings#assets`);
```

Comment:
Will need to change the hash location once [vsl-131](https://github.com/DapperCollectives/VESSEL/pull/96) is merged since this pr breaks the asset into two parts